server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java

```diff
@@ -134,7 +135,17 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
 
             final FileItemFactory fileItemFactory = new DiskFileItemFactory();
             final ServletFileUpload serviceFileUpload = createServletFileUpload(fileItemFactory);
-            final List<FileItem> items = serviceFileUpload.parseRequest(request);
+            if (checkUploadedFileSize) {
+                setUploadSizeMax(serviceFileUpload, request);
+            }
+            List<FileItem> items;
+            try {
+                items = serviceFileUpload.parseRequest(request);
+            } catch (final OutOfMemoryError e) {
+                throw new SizeLimitExceededException("The file exceeds its maximum permitted size.",
```

Comment:
there is no derefencing of object after the catch of the OOM. The JVM might not be able to have enough meomry to step back to a working state. Maybe try to set serviceFileUpload to null and call a GC would be safer (but would it be enough? Who is handling the big chunk of memory?)